opal/mca/threads/qthreads/threads_qthreads_mutex.h

```diff
@@ -42,46 +43,67 @@
 
 BEGIN_C_DECLS
 
-typedef opal_atomic_lock_t opal_thread_internal_mutex_t;
+typedef qthread_spinlock_t opal_thread_internal_mutex_t;
 
-#define OPAL_THREAD_INTERNAL_MUTEX_INITIALIZER           OPAL_ATOMIC_LOCK_INIT
-#define OPAL_THREAD_INTERNAL_RECURSIVE_MUTEX_INITIALIZER OPAL_ATOMIC_LOCK_INIT
+#define OPAL_THREAD_INTERNAL_MUTEX_INITIALIZER           QTHREAD_MUTEX_INITIALIZER
+#define OPAL_THREAD_INTERNAL_RECURSIVE_MUTEX_INITIALIZER QTHREAD_RECURSIVE_MUTEX_INITIALIZER
 
 static inline int opal_thread_internal_mutex_init(opal_thread_internal_mutex_t *p_mutex,
                                                   bool recursive)
 {
-    opal_atomic_lock_init(p_mutex, 0);
+    opal_threads_ensure_init_qthreads();
+    #if OPAL_ENABLE_DEBUG
+    int ret = qthread_spinlock_init(p_mutex,recursive);
+    if (QTHREAD_SUCCESS != ret) {
+        opal_output(0, "opal_thread_internal_mutex_init()");
+    }
+    #else
+    qthread_spinlock_init(p_mutex,recursive);
+    #endif
     return OPAL_SUCCESS;
 }
 
 static inline void opal_thread_internal_mutex_lock(opal_thread_internal_mutex_t *p_mutex)
 {
     opal_threads_ensure_init_qthreads();
```

Comment:
if lock requires the mutex to be initialized, then the `opal_threads_ensure_init_qthreads` has already been called and qthreads are definitively up and running.